src/app/modules/main/chat_section/module.nim

```diff
@@ -399,4 +400,9 @@ method onContactBlocked*(self: Module, publicKey: string) =
 
 method onContactDetailsUpdated*(self: Module, publicKey: string) =
   let (chatName, chatImage, isIdenticon) = self.controller.getOneToOneChatNameAndImage(publicKey)
+  let contact = self.controller.getContactById(publicKey)
+  if (contact.hasAddedUs and not contact.added):
+    let item = self.createItemFromPublicKey(publicKey)
```

Comment:
@jrainville I know that what I am going to say is not part of the changes made here, but I noticed that now, so in the method `createItemFromPublicKey` we have two calls to the contacts service:
```
proc createItemFromPublicKey(self: Module, publicKey: string): contacts_item.Item =
  let contact =  self.controller.getContact(publicKey)
  let (name, image, isIdenticon) = self.controller.getContactNameAndImage(contact.id)
  
  return contacts_item.initItem(contact.id, name, image, isIdenticon, contact.isContact(), contact.isBlocked(), contact.requestReceived())
```
but instead we can make it simpler, since we already have convenient method in contacts service named `getContactDetails`, so if you want you may add it to the controller and use in `createItemFromPublicKey` like this:
```
proc createItemFromPublicKey(self: Module, publicKey: string): contacts_item.Item =
  let contactDetails = self.controller.getContactDetails(publicKey)
  
  return contacts_item.initItem(contactDetails.details.id, contactDetails.displayName, contactDetails.icon, contactDetails.isIdenticon, contactDetails.details.isContact(), contactDetails.details.isBlocked(), contactDetails.details.requestReceived())
```